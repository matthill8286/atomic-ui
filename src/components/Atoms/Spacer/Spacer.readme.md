## How to use?
Use the spacer as a element to bring distance between two elements. 
The spacer is mobile first and the value is depending on the variables `spacing.baseMobile` for mobile and `spacing.base` for devices => tablet (see `src/styles/sc-vars-global.ts`).

