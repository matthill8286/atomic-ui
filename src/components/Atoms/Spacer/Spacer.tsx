import React from 'react'
import { media } from '@/styles/media'
import { css, styled } from '@/styles/styled'

export type SpacerDirection = 'horizontal' | 'vertical'

interface SpacerProps {
  size: string
  direction?: SpacerDirection
}

interface StyledSpacerProps {
  size: string
  direction: SpacerDirection
}

const StyledSpacer = styled.div<StyledSpacerProps>`
  display: block;
  width: 100%;

  ${({ direction, size, theme }) =>
    direction === 'vertical'
      ? css`
          width: ${theme.spacing.baseMobile[size]};
        `
      : css`
          height: ${theme.spacing.baseMobile[size]};

          ${media.md} {
            height: ${theme.spacing.base[size]};
          }
        `}
`

export const Spacer: React.FC<SpacerProps> = ({
  size,
  direction = 'horizontal',
  ...otherProps
}) => {
  return size ? <StyledSpacer size={size} direction={direction} {...otherProps} /> : null
}
