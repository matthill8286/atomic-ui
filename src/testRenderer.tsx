import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { saiyanTheme } from '@/styles/sc-vars-saiyan'
import { Theme } from '@/types'

export function renderWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>,
  theme = saiyanTheme
) {
  return renderer.create(<ThemeProvider theme={theme}>{tree}</ThemeProvider>).toJSON()
}

export function renderWithThemeAndRouter(tree: React.ReactElement, theme: Theme = saiyanTheme) {
  return renderer
    .create(
      <ThemeProvider theme={theme}>
        <Router>{tree}</Router>
      </ThemeProvider>
    )
    .toJSON()
}

export function renderWithRouter(tree: React.ReactElement) {
  return renderer.create(<Router>{tree}</Router>).toJSON()
}

export function mountWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(<ThemeProvider theme={saiyanTheme}>{tree}</ThemeProvider>)
}

export function mountWithThemeAndRouter<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(
    <ThemeProvider theme={saiyanTheme}>
      <Router>{tree}</Router>
    </ThemeProvider>
  )
}
