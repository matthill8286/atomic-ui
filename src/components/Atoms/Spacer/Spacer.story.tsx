import { storiesOf } from '@storybook/react'
import React from 'react'
import { FlexBox } from '@/components/Helper/FlexBox'
import { styled } from '@/styles/styled'
import { Spacer } from './Spacer'
import Readme from './Spacer.readme.md'

const StyledBlock = styled.div`
  height: 20px;
  width: 100%;
  background: red;
  float: left;
`

const StyledVerticalBlock = styled.div`
  height: 100px;
  width: 20px;
  background: red;
  float: left;
`

storiesOf('Design System/Atoms/Spacer', module)
  .add(
    'Default',
    () => (
      <>
        <StyledBlock />
        <Spacer size={'md'} />
        <StyledBlock />
        <Spacer size={'xxxl'} />
        <StyledBlock />
      </>
    ),
    {
      info: Readme,
    }
  )
  .add(
    'Vertical',
    () => (
      <FlexBox flexDirection="row">
        <StyledVerticalBlock />
        <Spacer direction="vertical" size={'md'} />
        <StyledVerticalBlock />
        <Spacer direction="vertical" size={'xxxl'} />
        <StyledVerticalBlock />
      </FlexBox>
    ),
    {
      info: Readme,
    }
  )
