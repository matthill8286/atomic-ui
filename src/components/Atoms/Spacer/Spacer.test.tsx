import React from 'react'
import { renderWithTheme } from '@/testRenderer'
import { Spacer } from './Spacer'

describe('Spacer', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(<Spacer size={'md'} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly in vertical direction', () => {
    const tree = renderWithTheme(<Spacer direction="vertical" size={'md'} />)
    expect(tree).toMatchSnapshot()
  })
})
