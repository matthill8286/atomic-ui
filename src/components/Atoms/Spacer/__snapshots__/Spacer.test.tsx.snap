// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer renders correctly 1`] = `
.c0 {
  display: block;
  width: 100%;
  height: 16px;
}

@media (min-width:752px) {
  .c0 {
    height: 24px;
  }
}

<div
  className="c0"
  direction="horizontal"
  size="md"
/>
`;

exports[`Spacer renders correctly in vertical direction 1`] = `
.c0 {
  display: block;
  width: 100%;
  width: 16px;
}

<div
  className="c0"
  direction="vertical"
  size="md"
/>
`;
